pkgs/stdenv/darwin/make-bootstrap-tools.nix

```diff
@@ -221,26 +265,36 @@ rec {
 
     buildCommand = ''
       mkdir -p $out/nix-support
-      echo "file tools ${build}/on-server/bootstrap-tools.nar.xz" >> $out/nix-support/hydra-build-products
+      for tool in ${build}/on-server/*; do
+        name=$(basename "$tool")
+        if [[ $name == bootstrap-tools* ]]; then
+          name=tarball
+        fi
+        echo "file $name $tool" >> $out/nix-support/hydra-build-products
+      done
     '';
   };
 
   bootstrapFiles = {
-    tools = "${build}/pack";
+    bash = "${build}/on-server/bash";
+    cpio = "${build}/on-server/cpio";
+    mkdir = "${build}/on-server/mkdir";
+    xz = "${build}/on-server/xz";
+    tarball = "${build}/on-server/bootstrap-tools.cpio.xz";
   };
```

Comment:
Does it go against the spirit of #202347 and the unmerged https://github.com/stephank/nixpkgs/commit/7be022c2f9e9a68c07cefb803612393d7b6849db, of providing a single file in the next bootstrap files bump?

Reply:
maybe!
is the unpacked nar archive writable? i thought that it would be readonly and there are a few minor fixups that take place.
- updating the library name for dylibs
- adding scripts for egrep, fgrep, etc that want a correct path in the shebang

it looks like @stephank was able to write to the archive. thanks for the link, i will try. probably save 1-2MB and a few lines of code and is also slicker.

Reply:
I guess so after `cp -R $tools $out && chmod -R u+w $out` in the commit.

Reply:
ah, in that case it doesn't seem like a win. the uncompressed archive is like 450 mb. so when fetchurl unpacks for us we then have to copy the uncompressed data (450MB) and then perform updates.

providing the 4 tools (bash, cpio, xz, mkdir) we save 450 MB of disk and an extra copy at the expense of needing to provide tools with no references into the nix store.

Reply:
Indeed, so `unpack = true;` should be dropped from that commit.

Reply:
one option would be to use a nar file that contains the compressed cpio archive and the 4 tools needed to extract and decompress the archive. This is not as nice as having the nix layer do all the work for us, but simplifies the bootstrap-tools download so it just requires one file rather than 5.

Reply:
ok -- i used a nar container to hold the compressed cpio archive and the exes needed to unpack it. 

(from PR https://github.com/NixOS/nixpkgs/pull/295558)

`bootstrapFiles` is set to the result of `<nix/fetchurl.nix>`
https://github.com/NixOS/nixpkgs/blob/30c4faca8d35a08bdf4c15ec9195867ce2b13310/pkgs/stdenv/darwin/bootstrap-files/x86_64-apple-darwin.nix#L1-L5

setup the derivation. PATH points to the exes needed to unpack and the path to the unpacked bin/
https://github.com/NixOS/nixpkgs/blob/30c4faca8d35a08bdf4c15ec9195867ce2b13310/pkgs/stdenv/darwin/default.nix#L51-L60